### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by Mkmittal**
**What this PR does / why we need it**:
Adds integration test
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Had to change a Prometheus dependency in the go mod.
**Release note**:

```improvement developer
Local IT for provider alicloud have been added
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @Mkmittal for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: test/integration/controller/controller_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package controller
```
Comment (himanshu-kun) [inline test/integration/controller/controller_suite_test.go:1]: add license

Comment (himanshu-kun) [reply test/integration/controller/controller_suite_test.go:1]: to all the files where not added

File: test/integration/controller/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		- apply secret resource for accesing the cloud provider service in the control cluster
+		- Create machineclass resource from file available in kubernetes directory of provider specific repo in control cluster
+	AfterSuite
+		- Delete the control and target clusters // As of now we are reusing the cluster so this is not required
+
+	Test: differentRegion Scheduling Strategy Test
+        1) Create machine in region other than where the target cluster exists. (e.g machine in eu-west-1 and target cluster exists in us-east-1)
+           Expected Output
+			 - should fail because no cluster in same region exists)
+
```
Comment (himanshu-kun) [inline test/integration/controller/controller_test.go:21]: remove this as we don't do this

File: test/integration/provider/alicloud.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		return nil, err
+	} else if len(machineList.MachineList) != 0 {
+		for _, machine := range machineList.MachineList {
+			machines = append(machines, machine)
+		}
+	}
+	return machines, nil
+}
+
+// getOrphanesInstances returns list of Orphan resources that couldn't be deleted
```
Comment (himanshu-kun) [inline test/integration/provider/alicloud.go:68]: change this docstring

File: test/integration/provider/alicloud.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	return instancesID, nil
+}
+
+// getOrphanesDisks returns list of Orphan disks
+func getOrphanedDisks(tagName string, tagValue string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	sess := newSession(machineClass, &v1.Secret{Data: secretData})
+	var volumeID []string
+	var tags = &[]ecs.DescribeDisksTag{{Key: tagName, Value: tagValue}}
+	input := ecs.CreateDescribeDisksRequest()
+	input.Tag = tags
```
Comment (himanshu-kun) [inline test/integration/provider/alicloud.go:94]: search for disks which are `available` state

File: test/integration/provider/alicloud.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+	for _, disk := range result.Disks.Disk {
+		volumeID = append(volumeID, disk.DiskId)
+	}
+	return volumeID, nil
+}
+
+// getOrphanedNICs returns list of Orphan NICs
+func getOrphanedNICs(tagName string, tagValue string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	sess := newSession(machineClass, &v1.Secret{Data: secretData})
+	var NICIDs []string
```
Comment (himanshu-kun) [inline test/integration/provider/alicloud.go:108]: unexport NICIDs, use nicIDs

File: test/integration/provider/rti.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	return nil
+}
+
+// probeResources will look for resources currently available and returns them
+func (r *ResourcesTrackerImpl) probeResources() ([]string, []string, []string, []string, error) {
+	// Check for VM instances with matching tags/labels
+	// Describe volumes attached to VM instance & delete the volumes
+	// Finally delete the VM instance
```
Comment (himanshu-kun) [inline test/integration/provider/rti.go:47]: why this comment?

File: test/integration/provider/rti.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+//If yes, then prints them and returns true. If not, then returns false
+func (r *ResourcesTrackerImpl) IsOrphanedResourcesAvailable() bool {
+	afterTestExecutionVMs, afterTestExecutionAvailDisks, afterTestExecutionAvailmachines, afterTestExecutionNICs, err := r.probeResources()
+	if err != nil {
+		fmt.Printf("Error probing orphaned resources: %s", err.Error())
+		return true
+	}
+
+	if afterTestExecutionVMs != nil || afterTestExecutionAvailDisks != nil || afterTestExecutionAvailmachines != nil || afterTestExecutionNICs != nil {
+		fmt.Printf("The following resources are orphans ... trying to delete them \n")
```
Comment (himanshu-kun) [inline test/integration/provider/rti.go:84]: `waiting till they get deleted or timeout happens` should be used

**Review by himanshu-kun — CHANGES_REQUESTED** [review]

File: test/integration/controller/controller_suite_test.go (hunk 1 above)
Comment (Mkmittal) [reply test/integration/controller/controller_suite_test.go:1]: done

File: Makefile [hunk 8]
```diff
@@ -18,9 +18,21 @@ IMAGE_REPOSITORY    := eu.gcr.io/gardener-project/gardener/machine-controller-ma
 IMAGE_TAG           := $(shell cat VERSION)
 PROVIDER_NAME       := alicloud
 PROJECT_NAME        := gardener
-CONTROL_NAMESPACE   := default
-CONTROL_KUBECONFIG  := dev/target-kubeconfig.yaml
-TARGET_KUBECONFIG   := dev/target-kubeconfig.yaml
+CONTROL_NAMESPACE   := shoot--i544024--it-testing
+CONTROL_KUBECONFIG  := dev/kubeconfigs/kubeconfig_control.yaml
+TARGET_KUBECONFIG   := dev/kubeconfigs/kubeconfig_target.yaml
+
```
[the remark this answers is not in the document]
Comment (Mkmittal) [reply Makefile]: done

File: test/integration/controller/controller_test.go (hunk 2 above)
Comment (Mkmittal) [reply test/integration/controller/controller_test.go:21]: done

File: test/integration/provider/alicloud.go (hunk 3 above)
Comment (Mkmittal) [reply test/integration/provider/alicloud.go:68]: done

File: test/integration/provider/alicloud.go (hunk 4 above)
Comment (Mkmittal) [reply test/integration/provider/alicloud.go:94]: done

File: test/integration/provider/alicloud.go (hunk 5 above)
Comment (Mkmittal) [reply test/integration/provider/alicloud.go:108]: done

File: test/integration/provider/rti.go (hunk 7 above)
Comment (Mkmittal) [reply test/integration/provider/rti.go:84]: done

File: test/integration/provider/rti.go (hunk 6 above)
Comment (Mkmittal) [reply test/integration/provider/rti.go:47]: by mistake, removed.

File: test/integration/provider/alicloud.go [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+// getOrphanedInstances returns list of Orphan resources that couldn't be deleted.
+func getOrphanedInstances(tagName string, tagValue string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	sess := newSession(machineClass, &v1.Secret{Data: secretData})
+	var instancesID []string
+	var tags = &[]ecs.DescribeInstancesTag{{Key: tagName, Value: tagValue}}
+	input := ecs.CreateDescribeInstancesRequest()
+	input.InstanceName = "instance-state-name"
+	input.Status = "running"
+	input.Tag = tags
+
```
Comment (shaoyongfeng) [inline test/integration/provider/alicloud.go:92]: When and where these orphan resources are created? 
why instance name is hardcoded.

Comment (Mkmittal) [reply test/integration/provider/alicloud.go:92]: These orphan resources are/might get created during the Integration test. Also, the instance name was there when I tested something and forgot to remove it.
Thanks, I will be removing this, as it is not required.

File: test/integration/provider/alicloud.go [hunk 10]
```diff
@@ -80,7 +80,7 @@ func getMachines(machineClass *v1alpha1.MachineClass, secretData map[string][]by
 	return machines, nil
 }
 
-// getOrphanesInstances returns list of Orphan resources that couldn't be deleted
+// getOrphanedInstances returns list of Orphan resources that couldn't be deleted.
```
Comment (himanshu-kun) [inline test/integration/provider/alicloud.go:84]: it should be 
`getOrphanedInstances returns list of Orphan resources`

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
looks good just do the docstring change requested

File: test/integration/provider/alicloud.go (hunk 10 above)
Comment (Mkmittal) [reply test/integration/provider/alicloud.go:84]: ok

Comment (Mkmittal) [reply test/integration/provider/alicloud.go]: done

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
could you also add the `test-integration` make rule in the Makefile

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**